Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/GeckoCodeWidget.h"
+
+#include <QFontDatabase>
+#include <QFormLayout>
+#include <QLabel>
+#include <QListWidget>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QTextEdit>
+#include <QVBoxLayout>
+
+#include "Common/FileUtil.h"
+#include "Common/IniFile.h"
+#include "Core/ConfigManager.h"
+#include "Core/GeckoCodeConfig.h"
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+#include "DolphinQt2/GameList/GameFile.h"
+
+GeckoCodeWidget::GeckoCodeWidget(const GameFile& game)
+    : m_game(game), m_game_id(game.GetGameID().toStdString()), m_game_revision(game.GetRevision())
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  auto game_ini_default = SConfig::GetInstance().LoadDefaultGameIni(m_game_id, m_game_revision);
+  IniFile game_ini_local;
+  game_ini_local.Load(File::GetUserPath(D_GAMESETTINGS_IDX) + m_game_id + ".ini");
+
+  m_gecko_codes = Gecko::LoadCodes(game_ini_default, game_ini_local);
+
+  UpdateList();
+}
+
+void GeckoCodeWidget::CreateWidgets()
+{
+  auto* layout = new QVBoxLayout;
+  m_warning = new CheatWarningWidget(m_game_id);
+  m_code_list = new QListWidget;
+  m_name_label = new QLabel;
+  m_creator_label = new QLabel;
+  m_code_description = new QTextEdit;
+  m_code_view = new QTextEdit;
+  m_download_codes = new QPushButton(tr("Download Codes (WiiRD Database)"));
+
+  QFont monospace(QStringLiteral("Monospace"));
+
+  if (monospace.family() != QStringLiteral("Monospace"))
```

Comment:
1) This is a tautology. The `family()` method returns what is passed in to the constructor.
2) Why not always use the system font? Does it break on some systems? If so, please add a comment.
3) When setting the font to the system font, it should be constructed from the system font's family, rather than copied directly. The system font has a size attached that isn't always appropriate (e.g., it's too small here), so it's better to let Qt figure that out.

Reply:
1. Sorry, thought it might change with Qt trying to automatically finding the most fitting font.
2. It breaks on my system :( (Linux x86_64, Qt 5.7)
3. Okay, didn't know that either. Will change it.